### ISSUE TITLE
no-use-before-declare false positive with destructured parameters

### ISSUE THREAD

**Issue by cartant** [palantir/tslint#3906]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.8.3
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
function f({ name: optionalName }: { name?: string }): void {}
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "rules": {
    "no-use-before-declare": { "severity": "error" }
  }
}
```

#### Actual behavior

Effects a `variable 'name' used before declaration` error.

#### Expected behavior

No error.

**Comment by giladgray** [comment]
cannot reproduce. please try with latest version and file a new issue if it still occurs.
